clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareCompletedFetch.java

```diff
@@ -99,10 +99,17 @@ public class ShareCompletedFetch {
     }
 
     private List<OffsetAndDeliveryCount> buildAcquiredRecordList(List<ShareFetchResponseData.AcquiredRecords> partitionAcquiredRecords) {
-        List<OffsetAndDeliveryCount> acquiredRecordList = new LinkedList<>();
+        List<OffsetAndDeliveryCount> acquiredRecordList = new ArrayList<>();
```

Comment:
By default, a new list will have space for 10 elements. Resizing is expensive. Maybe one optimisation would be to see how many offsets are in the first element in the `partitionAcquiredRecords`, and using that number as the initial size of the list. In the case of only one batch of offsets, the list will be the correct size alrready. wdyt?